### PR TITLE
Zero view factors between conditioned basement surfaces and living surfaces

### PR DESCRIPTION
**Pull request by yzhou601**
- [x] Test OS SDK of wrapping up "👀 factor" object
- [x] Implement E+ approximate view factor algorithm
- [x] Implement the whole workflow from calculating approximate view factor numbers to running OS simulation
- [x] Check results of implementation
- [ ] Request review & modify code

### CODE REVIEW

**Comment by yzhou601** [comment]
The implemented approximate view factors(output by E+) looks correct. So OS interpretation should be good.

**Comment by yzhou601** [comment]
This PR failed the CI by a partial HVAC check, only `base-hvac-air-to-air-heat-pump-2-speed-33percent.xml` and `base-hvac-air-to-air-heat-pump-var-speed-33percent.xml` went off tolerance (5%). I checked load numbers they are the same as base whole system, so I guess it's not relevant to this PR (impacted more on load instead of systems). Not sure if it's ok to relax a bit of tolerance? @shorowit

**Comment by shorowit** [comment]
@yzhou601 You should now be able to compare CI results for [this branch](https://circleci.com/gh/NREL/OpenStudio-HPXML/1086) and [master](https://circleci.com/gh/NREL/OpenStudio-HPXML/1083). Can you look them over?

**Comment by shorowit** [comment]
For what it's worth, I pulled these changes over to the ERI repo and the ASHRAE 140 basement cases now show good agreement with older results using the 2-zone model. (All the non-basement cases are correctly unchanged.)

image

It's only one building in one location, but it's encouraging!

**Comment by yzhou601** [comment]
@shorowit I think it's hard to judge in which scale the view factors are negligible, eg. for a model with hundreds of surfaces, 0.01 or 0.001 might not mean "tiny", though in that case the performance issue will be more significant and assuming a limitation will give more performance gain.

File: measure.rb [hunk 1]
```diff
@@ -829,15 +829,12 @@ def self.calc_approximate_view_factor(runner, model, all_surfaces)
...
               # calculate surface and its sub surfaces view factors
-              parent_surface_a = surface2.grossArea
+              parent_surface_a = surface2.netArea
+              if parent_surface_a > 0.01 # base surface of a sub surface: window/door etc.
+                surface_vf_map[surface2] = parent_surface_a / zone_seen_area
+              end
               surface2.subSurfaces.each do |sub_surface|
-                parent_surface_a -= sub_surface.grossArea
-                if parent_surface_a < 0.01 # base surface of a sub surface: window/door etc.
-                  surface_vf_map[sub_surface] = surface2.grossArea / zone_seen_area
```
Comment (yzhou601) [inline measure.rb]: This line is not correct if it has more than one sub surfaces, I reverted sub surface view factor calculation the same as before.

**Pull request merged**